### ISSUE TITLE
String -> File coercion issues with "read_string()" function

### ISSUE THREAD

**Issue by ruchim** [broadinstitute/cromwell#3065]
Given this workflow:
```
workflow w {
	call t {}
	output { t.* }
}

task t {
	
	command {
		touch file.txt
		echo "file.txt" > filename.txt
	}

	output {
		File f1 = "filename.txt"
		File f2 = read_string(f1)		
	}

	runtime {
	  docker: "ubuntu"
	}
}
```
One would expect 2 files to be marked for delocalization for `call t`, yet only "filename.txt" is properly localized, but not "file.txt". This behavior is only observed when running against the JES backend, and works as expected on the SharedFileSystem backend. This is a regression introduced in Cromwell v30.

When checking Google's operation metadata, the outputs are listed as:
```
outputs:
     filename.txt: gs://rm-dev/w/6b29d52a-d665-4ce9-b042-8cd4e4374473/call-t/filename.txt
     t-rc.txt: gs://rm-dev/w/6b29d52a-d665-4ce9-b042-8cd4e4374473/call-t/t-rc.txt
```

**Comment by ruchim** [comment]
Just to clarify -- this isn't currently not possible on the JES backend, only Local. However, Pipelines APIv2 would allow for more flexible delocalization.

**Comment by ruchim** [comment]
@tjeandet —out of curiosity, should this already work in the current v2 backend?

**Comment by Horneth** [comment]
No, although we've briefly discussed it with Chris and Miguel I think. We already implemented similar logic for CWL but not for WDL (which will admittedly be a little more complicated). There's quite a few other cases like that that we can now cover in Papi2, basically any file/directory to delocalize that can't be determined statically before the job is submitted we couldn't support in v1 but could now in v2.
